docs/DeveloperGuide.md

```diff
@@ -1,34 +1,330 @@
+    1.4. [Main Features](#14-main-features)
+1. [Design](#2-design)<br>
+    2.1. [Architecture](#21-architecture)<br>
+    2.2. [Task Component](#22-task-component)<br>
+    2.3. [Study Area Component](#23-study-area-component)<br>
+1. [Implementation](#3-implementation)<br>
+    3.1. [Scheduling Tasks](#31-scheduling-tasks)<br>
+    3.2. [Listing Study Areas](#32-listing-study-areas)<br>
+1. [Testing](#4-testing)<br>
 
+[Appendix A: Product Scope](#appendix-a-product-scope)<br>
+[Appendix B: User Stories](#appendix-b-user-stories)<br>
+[Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)<br>
+[Appendix D: Glossary](#appendix-d-glossary)<br>
+[Appendix E: Instruction for Manual Testing](#appendix-e-instructions-for-manual-testing)<br>
 
-## Product Scope
-### Target user profile
+## 1. Introduction
 
-{Describe the target user profile}
+### 1.1. Purpose
 
-### Value proposition
+The document is meant to teach you on how OrgaNice! was developed and the design and implementations behind the software.
+With this document, you should have a better understanding of the framework of the application. 
 
-{Describe the value proposition: what problem does it solve?}
+### 1.2. Target Reader
 
-## User Stories
+The target reader of this developer guide are:
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+- Project Managers
+- Future Developers
+- Build Verification Testers
 
-## Non-Functional Requirements
+### 1.3. Brief Description
 
-{Give non-functional requirements}
+OrgaNice! is a task manager integrated with a Study Area search function. You can use it to manage your tasks, be it events or deadlines. On top of that, 
+it is capable of assisting students in finding Study Areas that meets their desired criteria.
 
-## Glossary
+### 1.4. Main Features
 
-* *glossary item* - Definition
+1. Scheduling Tasks
+    - You can manage tasks based on priority for events and countdown for deadlines.
+1. Study Area. For more information of the design and implementation for this feature, click [here](#31-scheduling-tasks)
+    - You can find a Study Area that meets your criteria. The software however have a limited number of supported 
+    criteria. For more information of the design and implementation for this feature, click [here](#32-listing-study-areas)
 
-## Instructions for Manual Testing
+## 2. Design   
+ {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}    
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+[comment]: # (@@author GanapathySanathBalaji)
+### 2.1. Architecture  
+  
+![Architecture](images/Architecture.png)
```

Comment:
Good use of Architecture diagram to get an overview of the Software Architecture.